### PR TITLE
fix identified ASAN problem

### PR DESCRIPTION
**Pull request by skinnari**
#### PR description:

- fix for issue #30913 

#### PR validation:

nbr of tracks identified before/after fix is identical in ttbar PU=200 events

### CODE REVIEW

**Comment by mrodozov** [comment]
please test
any reason not to keep the 
`indexArray[0] < matches1.size()`
in the body of the `while` before the `&&` as in #31555 ? 
And also did you try to run this with any of the failing relvals/unittests ?
I think this if(index)

File: L1Trigger/TrackFindingTracklet/src/FitTrack.cc [hunk 1]
```diff
@@ -934,10 +934,12 @@ void FitTrack::execute() {
     bool match = false;
 
     if (indexArray[0] < matches1.size()) {
-      while (matches1[indexArray[0]] == bestTracklet && indexArray[0] < matches1.size()) {
+      while (matches1[indexArray[0]] == bestTracklet) {
```
Comment (smuzaffar) [inline L1Trigger/TrackFindingTracklet/src/FitTrack.cc:937]: I think something like the following is enough and there
```
while (indexArray[0] < matches1.size() && matches1[indexArray[0]] == bestTracklet) {
```

File: L1Trigger/TrackFindingTracklet/src/FitTrack.cc [hunk 2]
```diff
@@ -934,10 +934,12 @@ void FitTrack::execute() {
...
         indexArray[0]++;
         nMatches++;
         match = true;
+        if (indexArray[0] >= matches1.size())
```
Comment (smuzaffar) [inline L1Trigger/TrackFindingTracklet/src/FitTrack.cc:941]: this should not be needed is the condition goes in the `while`

File: L1Trigger/TrackFindingTracklet/src/FitTrack.cc [hunk 3]
```diff
@@ -934,10 +934,12 @@ void FitTrack::execute() {
     bool match = false;
 
     if (indexArray[0] < matches1.size()) {
```
Comment (smuzaffar) [inline L1Trigger/TrackFindingTracklet/src/FitTrack.cc:936]: once indexArray[0] < matches1.size() is part of `while` then this is also not needed

**Comment by smuzaffar** [comment]
@skinnari , I mean something like the following should be enough
```
-    if (indexArray[0] < matches1.size()) {
-      while (matches1[indexArray[0]] == bestTracklet && indexArray[0] < matches1.size()) {
-        indexArray[0]++;
-        nMatches++;
-        match = true;
-      }
+    while (indexArray[0] < matches1.size() && matches1[indexArray[0]] == bestTracklet) {
+      indexArray[0]++;
+      nMatches++;
+      match = true;
```

**Comment by smuzaffar** [comment]
@skinnari , I would suggest to close this PR in favor of https://github.com/cms-sw/cmssw/pull/31555

**Pull request closed without merging**